mitosheet/src/components/endo/celleditor/CellEditor.tsx

```diff
@@ -454,28 +370,40 @@ const CellEditor = (props: {
 
         const columnID = props.sheetData.data[props.editorState.columnIndex].columnID;
         const columnHeader = props.sheetData.data[props.editorState.columnIndex].columnHeader;
-        const index = props.sheetData.index[props.editorState.rowIndex];
+        const rowIndex = props.sheetData.index[props.editorState.rowIndex];
         const formula = getFullFormula(props.editorState.formula, columnHeader, props.editorState.pendingSelectedColumns)
 
         // Mark this as loading
         setLoading(true);
         
         let errorMessage: MitoError | undefined = undefined;
+
         // Make sure to send the write type of message, depending on the editor
-        if (isFormulaColumn) {
-            errorMessage = await props.mitoAPI.editSetColumnFormula(
-                props.sheetIndex,
-                columnID,
-                formula
-            )
+        if (rowIndex >= 0) {
+            if (isFormulaColumn) {
+                // Change of formula
+                errorMessage = await props.mitoAPI.editSetColumnFormula(
+                    props.sheetIndex,
+                    columnID,
+                    formula,
+                    props.editorState.editorLocation
+                )
+            } else {
+                // Change of data
+                errorMessage = await props.mitoAPI.editSetCellValue(
+                    props.sheetIndex,
+                    columnID,
+                    rowIndex,
+                    formula,
+                    props.editorState.editorLocation
+                )
+            } 
         } else {
-            errorMessage = await props.mitoAPI.editSetCellValue(
-                props.sheetIndex,
-                columnID,
-                index,
-                formula
-            )
-        } 
+            // Change of column header
```

Comment:
Can we move this up to the first condition? E.g. check if rowIndex == -1?

It sounds kinda silly, but I would expect the if statements to flow in the same order that the sheet data does - like column header before everything else. My brain took a second to process as it is - I was expecting column headers first!